src/io/flutter/editor/outline/OpenEditorOutlineService.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright 2019 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.editor.outline;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.intellij.openapi.Disposable;
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.editor.ex.EditorEx;
+import com.intellij.openapi.fileEditor.*;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.io.FileUtil;
+import com.intellij.openapi.vfs.VirtualFile;
+import io.flutter.FlutterUtils;
+import io.flutter.dart.FlutterDartAnalysisServer;
+import io.flutter.dart.FlutterOutlineListener;
+import org.dartlang.analysis.server.protocol.FlutterOutline;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+import java.util.*;
+
+/**
+ * Service that watches the Flutter Outlines for all open Dart files.
```

Comment:
Note this isn't really all open files. Just all Dart files that are visible to the user. Files that are open but in tabs that are are not selected are not included which is a good thing for performance.